apps/mobile/src/components/DrawerNavigation/index.tsx

```diff
@@ -0,0 +1,50 @@
+import { StyleSheet } from 'react-native';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import { useSafeAreaInsets } from '@walless/app';
+import { mockWidgets } from '@walless/engine';
+import ExploreScreen from 'screens/Dashboard/Explore';
+import WidgetScreen from 'screens/Dashboard/Home/Widget';
+
+import Sidebar, { sidebarWidth } from './Sidebar';
+
+const Drawer = createDrawerNavigator();
+
+export const DrawerNavigation = () => {
+	const { top } = useSafeAreaInsets();
+
+	return (
+		<Drawer.Navigator
+			drawerContent={Sidebar}
+			screenOptions={{
+				headerShown: false,
+				drawerStyle: [
+					styles.drawer,
+					{
+						paddingTop: top,
+						backgroundColor: '#131C24',
+					},
+				],
+				overlayColor: 'transparent',
+			}}
+			backBehavior="order"
+			initialRouteName="Explore"
+		>
+			<Drawer.Screen name="Explore" component={ExploreScreen} />
+			{mockWidgets.map((widget) => (
+				<Drawer.Screen
+					key={widget._id}
+					name={widget._id}
+					component={WidgetScreen}
+				/>
+			))}
```

Comment:
we don't want to pre-render all Widgets using Navigation system, have a look at [this reference](https://github.com/cocrafts/walless/blob/5d907b40c550f5bcf267f37d70231556f924d6ce/apps/web/src/screens/Dashboard/Embed.tsx#L13).